core/model/step/dropoff/dropoff.go

```diff
@@ -0,0 +1,91 @@
+package dropoff
+
+import (
+	"context"
+	"time"
+)
+
+type State int
+type Event int
+
+const (
+	New State = iota
+	Ready
+	Completed
+	Cancelled
+)
+
+func (s State) String() string {
```

Comment:
This is a bad practice. First, it allocates on every call. Second, you don't check for boundaries. Use simple switch.